pennylane/ops/qubit/state_preparation.py

```diff
@@ -88,6 +91,24 @@ def compute_decomposition(n, wires):
         """
         return [BasisStatePreparation(n, wires)]
 
+    def matrix(self, wire_order=None):
+        """Returns a ket vector representing the state being created."""
+        if wire_order is None:
+            num_wires = len(self.wires)
+            indices = self.parameters[0]
+        else:
+            wires = Wires(wire_order)
+            if not wires.contains_wires(self.wires):
+                raise WireError("Custom wire_order must contain all BasisState wires")
+            num_wires = len(wires)
+            indices = [0] * num_wires
+            for wire_idx, value in zip(self.wires, self.parameters[0]):
+                indices[wires.index(wire_idx)] = value
+
+        ket = np.zeros(2**num_wires, dtype="complex128")
```

Comment:
this is a relic from the previous PR I had with `initialize_state`, because tensorflow doesn't like the `complex` type:
```pycon
>>> qml.math.zeros(4, dtype=complex, like="tensorflow")
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/Users/matthews/.pyenv/versions/pennylane/lib/python3.9/site-packages/autoray/autoray.py", line 85, in do
    return get_lib_fn(backend, fn)(*args, **kwargs)
  File "/Users/matthews/.pyenv/versions/pennylane/lib/python3.9/site-packages/tensorflow/python/util/traceback_utils.py", line 153, in error_handler
    raise e.with_traceback(filtered_tb) from None
  File "/Users/matthews/.pyenv/versions/pennylane/lib/python3.9/site-packages/tensorflow/python/framework/dtypes.py", line 762, in as_dtype
    raise TypeError(f"Cannot convert the argument `type_value`: {type_value!r} "
TypeError: Cannot convert the argument `type_value`: <class 'complex'> to a TensorFlow DType.
>>> qml.math.zeros(4, dtype="complex128", like="tensorflow")
<tf.Tensor: shape=(4,), dtype=complex128, numpy=array([0.+0.j, 0.+0.j, 0.+0.j, 0.+0.j])>
```

Are you proposing that I don't specify any type at all, and let it be inferred from the calling context?